lib/super_digit.rb

```diff
@@ -1,15 +1,25 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def super_digit(n)
-   
-end
-  
+  return n if n <= 9
+  return super_digit(n.digits.sum) if n > 9
+end 
 
-# Time Complexity - ?
-# Space Complexity - ?
-def refined_super_digit(n, k)
-    
+# Time Complexity - O(n+m) where m = k
+# Space Complexity - O(n)
+def refined_super_digit(n, k)  
```

Comment:
This is the brute force solution.  Can you think of a way to do this by finding the super_digit(n) and using it?